builtin/logical/aws/path_static_creds_test.go

```diff
@@ -1 +1,90 @@
 package aws
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	"github.com/fatih/structs"
+
+	"github.com/hashicorp/vault/sdk/framework"
+
+	"github.com/hashicorp/vault/sdk/logical"
+)
+
+func TestStaticCredsRead(t *testing.T) {
+	// setup
+	config := logical.TestBackendConfig()
+	config.StorageView = &logical.InmemStorage{}
+	bgCTX := context.Background() // for brevity later
+
+	// insert a cred to get
+	creds := &awsCredentials{
+		AccessKeyID:     "foo",
+		SecretAccessKey: "bar",
+	}
+	entry, err := logical.StorageEntryJSON(formatCredsStoragePath("test"), creds)
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = config.StorageView.Put(bgCTX, entry)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// cases
+	cases := []struct {
+		name             string
+		roleName         string
+		expectedError    error
```

Comment:
Do we still need this struct field and its check below?

Reply:
it is technically possible to return an error, so i wanted to leave it there for futureproofing